### PR TITLE
feat: add "new transaction" feature to wallet page

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Access "New transaction" feature / modal from the "Wallet" page with the selected account prefilled to current account (the account displayed on the detail page).

# Changes

- path `selectedAccount` to `NewTransactionModal`
- resolve identifier from route path
- get account for identifier from store
- add utils to resolve last detail of path as `string`
- display spinner while loading accounts on wallet page
- enabled - disabled action on wallet page until data are loaded
- bind "new transaction" action modal with button

# Screenshots

Capture d’écran 2022-04-25 à 10 48 48
Capture d’écran 2022-04-25 à 10 48 56
Capture d’écran 2022-04-25 à 10 49 02

### CODE REVIEW

File: frontend/svelte/src/routes/Wallet.svelte [hunk 1]
```diff
@@ -8,38 +8,67 @@
...
+
+  let accountIdentifier: string | undefined;
+  $: accountIdentifier = routePathAccountIdentifier($routeStore.path);
+
+  let mainAccount: Account | undefined;
+  $: mainAccount = $accountsStore?.main;
+
+  // TODO(L2-429): context and store for selectedAccount
+  let selectedAccount: Account | undefined;
+  $: accountIdentifier,
```
Comment (mstrasinskis) [inline frontend/svelte/src/routes/Wallet.svelte:41]: Is `accountIdentifier` needed here?

File: frontend/svelte/src/tests/routes/Accounts.spec.ts [hunk 2]
```diff
@@ -114,4 +115,38 @@ describe("Accounts", () => {
...
+      expect(
+        getByText(en.accounts.select_source, { exact: false })
+      ).toBeInTheDocument();
+    });
+  });
+
+  it("should open add account modal", async () => {
+    const { container, getByText } = render(Accounts);
+
+    const button = container.querySelector(
```
Comment (mstrasinskis) [inline frontend/svelte/src/tests/routes/Accounts.spec.ts:139]: getByTestId?
_#nitpick_

File: frontend/svelte/src/tests/routes/Wallet.spec.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    it("should render title", async () => {
+      const { getAllByText } = render(Wallet);
+
+      expect(getAllByText(en.wallet.title).length).toBeGreaterThan(0);
+    });
+
+    it("should render a spinner while loading", () => {
+      const { container } = render(Wallet);
+
+      expect(container.querySelector('[data-tid="spinner"]')).not.toBeNull();
```
Comment (mstrasinskis) [inline frontend/svelte/src/tests/routes/Wallet.spec.ts:51]: getByTestId?
_#nitpick_

File: frontend/svelte/src/tests/routes/Wallet.spec.ts [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        );
+    });
+
+    afterAll(() => jest.clearAllMocks());
+
+    it("should hide spinner when accounts are loaded", async () => {
+      const { container } = render(Wallet);
+
+      await waitFor(() =>
+        expect(container.querySelector('[data-tid="spinner"]')).toBeNull()
```
Comment (mstrasinskis) [inline frontend/svelte/src/tests/routes/Wallet.spec.ts:104]: getByTestId?
#nitpick

**Review by mstrasinskis — APPROVED** [review]
LGTM, just a couple of comments

File: frontend/svelte/src/routes/Wallet.svelte (hunk 1 above)
Comment (peterpeterparker) [reply frontend/svelte/src/routes/Wallet.svelte:41]: Yep, user can manually change the identifier in the browser url

File: frontend/svelte/src/tests/routes/Wallet.spec.ts (hunk 3 above)
Comment (peterpeterparker) [reply frontend/svelte/src/tests/routes/Wallet.spec.ts:51]: Thanks!

**Review by mstrasinskis — APPROVED** [review]
Thanks for description.

**Pull request merged**